src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -301,6 +320,28 @@ Image& Frame::image() const
     return *impl->m_image.get();
 }
 
+Image& Frame::ref_image() const
+{
+    return *impl->m_ref_image.get();
+}
+
+bool Frame::validate_ref_image() const
```

Comment:
Yeah, I wasn't fond of it for those reasons as well - your feedback and a second pass have helped simplify the way I'm thinking about it, I've pushed an update.

I think I was getting hung up on the studio workflow, and expected users to see meaningful output if, for instance, the user changes the resolution in studio and then wonders why the AOV returns blank. However, there isn't actually UI to even support selecting a reference images yet, if there was then it might be worth considering being more verbose.

Reply:
It's much better now!